File: libs/contracts/Dto/AppointmentDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace HSMS.contracts.Dto
{
    public class AppointmentDto
    {
        public int AppointmentId { get; set; }

        public int PatientId { get; set; }

        public string MedicalRecordNumber { get; set; } = string.Empty;
        public string FirstName { get; set; } = string.Empty;
        public string LastName { get; set; } = string.Empty;

        public int DoctorId { get; set; }

        public string DoctorName { get; set; }

        public DateTimeOffset AppointmentDateTime { get; set; }

        public string Status { get; set; } = string.Empty;

        public string? ReasonForVisit { get; set; }

        public DateTimeOffset CreatedAt { get; set; }
    }
}

File: apps/HSMS-fe/src/app/models/Appointment.dto.ts
export interface AppointmentData {
  appointmentId: number;
  patientId: number;
  medicalRecordNumber: string;
  firstName: string;
  lastName: string;
  doctorId: number;
  doctorName: string;
  appointmentDateTime: string;
  status: string;
  reasonForVisit: string;
  createdAt?: string;
}

export interface CreateAppointmentDto {
  patientId: number;
  doctorId: number;
  appointmentDateTime: string; // ISO 8601 string
  reasonForVisit: string;
}

export interface patientDropdown {
  id: number;
  patientname: string;

}

export interface DoctorOption {
  docId: number;
  docname: string;

}

export interface UpdateStatusDto {
  appointmentId: number;
  status: string;
}